### PR TITLE
Fix/nlsc

### PR DESCRIPTION
**Pull request by nathangibson14**
Changes Reich-Moore resonance reconstruction to use NLS instead of NLSC.  @whaeck also made some subtle changes to enable existing test to be retained, as adding empty spin groups is an important capability.

Reviewing this one is a bit odd, as both @whaeck and I have made changes.  I'll review the changes I didn't make, etc.

### CODE REVIEW

File: src/resonanceReconstruction/rmatrix/src/makeReichMooreChannelData.hpp [hunk 1]
```diff
@@ -124,12 +124,24 @@ makeReichMooreChannelData(
     auto rwFission2 = wFission2 | ranges::view::transform( toOtherWidth );
 
     // see how many channels we have
+    bool bElastic = not isAllZero( wElastic );
```
Comment (nathangibson14) [inline src/resonanceReconstruction/rmatrix/src/makeReichMooreChannelData.hpp]: This certainly works, but I don't like the naming conventions that led to `bElastic` and `cElastic` as variables.  I'd much rather see `elasticPopulated` or `hasElastic`, etc.

File: src/resonanceReconstruction/rmatrix/test/fromENDFResolvedReichMoore.test.hpp [hunk 2]
```diff
@@ -205,1806 +311,3187 @@ SCENARIO( "fromENDF - LRF3" ) {
...
+    " 2.511660+3 0.000000+0 3.123558-1 3.926596-2 1.002582+0 1.971943-29437 2151     \n"
+    " 2.514090+3 1.000000+0 1.322666-1 3.926598-2-2.281954-1 0.000000+09437 2151     \n"
+    " 2.516520+3 1.000000+0 1.436018-1 3.926556-2 3.888109-2 0.000000+09437 2151     \n"
+    " 2.518950+3 1.000000+0 1.662716-1 3.926594-2 2.224877-1 0.000000+09437 2151     \n"
+    " 2.521380+3 1.000000+0 1.600835-1 3.926564-2 8.565578-2 0.000000+09437 2151     \n"
+    " 2.523800+3 1.000000+0 1.829061-1 3.926666-2 4.074383-1 0.000000+09437 2151     \n"
+    "                                                                  9437 2  0     \n" );
+}
+
+std::string Pu239LRF3WithNLS4() {
```
Comment (nathangibson14) [inline src/resonanceReconstruction/rmatrix/test/fromENDFResolvedReichMoore.test.hpp]: If you don't think it's worth it, you can leave it as is, but...

It's really silly to have 1000 lines of the real Pu evaluation twice.  Can't you just turn the function into something that takes NLS as an argument?  Or use less than 1043 resonances to do this?

File: src/resonanceReconstruction/rmatrix/src/makeReichMooreChannelData.hpp (hunk 1 above)
Comment (whaeck) [reply src/resonanceReconstruction/rmatrix/src/makeReichMooreChannelData.hpp]: Good point, this used to be my old naming convention: b for boolean. I changed it to has... since it is clearer.

File: src/resonanceReconstruction/rmatrix/test/fromENDFResolvedReichMoore.test.hpp (hunk 2 above)
Comment (whaeck) [reply src/resonanceReconstruction/rmatrix/test/fromENDFResolvedReichMoore.test.hpp]: Done.

**Review by whaeck — DISMISSED** [review]

**Comment by whaeck** [comment]
@nathangibson14 I looked at your changes and approved those.
